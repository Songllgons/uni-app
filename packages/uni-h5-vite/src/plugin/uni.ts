import {
  type UniVitePlugin,
  transformH5BuiltInComponents,
  transformMatchMedia,
  transformPageHead,
  transformRefresherSlot,
  transformTapToClick,
  transformUniH5Jsx,
} from '@dcloudio/uni-cli-shared'
import { isH5CustomElement, isH5NativeTag } from '@dcloudio/uni-shared'
import type { CompilerOptions } from '@vue/compiler-core'

function realIsH5CustomElement(tag: string) {
  return isH5CustomElement(tag, process.env.UNI_APP_X === 'true')
}

export const compilerOptions: CompilerOptions = {
  isNativeTag: isH5NativeTag,
  isCustomElement: realIsH5CustomElement,
  nodeTransforms: [
    transformRefresherSlot,
    transformH5BuiltInComponents,
    transformTapToClick,
    transformMatchMedia,
    transformPageHead,
  ],
}

export function createUni(): UniVitePlugin['uni'] {
  return {
    copyOptions: {
      assets: ['hybrid/html/**/*', 'uni_modules/*/hybrid/html/**/*'],
    },
    compilerOptions,
    jsxOptions: {
      babelPlugins: [transformUniH5Jsx],
    },
  }
}
