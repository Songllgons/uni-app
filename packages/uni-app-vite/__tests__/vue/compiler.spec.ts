import { compileTemplate } from '@vue/compiler-sfc'
import { uniOptions } from '../../src/plugin/uni/index'
const { compilerOptions } = uniOptions('vue')!
const filename = 'foo.vue'

function compile(source: string) {
  return compileTemplate({
    source,
    filename,
    id: filename,
    compilerOptions: {
      ...compilerOptions,
    },
  })
}

function genCode(source: string) {
  return compile(source).code
}

const codes = [
  `<view :prop="options" :change:prop="renderScript.updateOptions"/>`,
]

describe('app-vue: compiler', () => {
  codes.forEach((code) => {
    test(code, () => {
      expect(genCode(code)).toMatchSnapshot()
    })
  })
})
