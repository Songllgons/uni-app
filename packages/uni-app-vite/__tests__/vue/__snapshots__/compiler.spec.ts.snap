// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-vue: compiler <view :prop="options" :change:prop="renderScript.updateOptions"/> 1`] = `
"import { wp as _wp, openBlock as _openBlock, createElementBlock as _createElementBlock } from "vue"

const _hoisted_1 = ["prop", "change:prop"]

export function render(_ctx, _cache) {
  return (_openBlock(), _createElementBlock("view", {
    prop: _wp(_ctx.options),
    "change:prop": _ctx.renderScript.updateOptions
  }, null, 8 /* PROPS */, _hoisted_1))
}"
`;
